PuzzleSlider/src/Engine/GameData.java

```diff
@@ -124,6 +137,23 @@ public GridPoint getEmptyTile() {
 		return mBlankSpot;
 	}
 	
+	public int getMoveCount() {
+		return mMoveCount;
+	}
+	
+	public int incrementMoveCount(){
+		mMoveCount++;
+		return mMoveCount;
```

Comment:
As our master Gyarfas always saying "What will be in the future ..." So rewrite ;)